tests/CommunityToolkit.Aspire.RavenDB.Client.Tests/AspireRavenDBExtensionsTests.cs

```diff
@@ -0,0 +1,410 @@
+
+        builder.AddKeyedRavenDBClient(serviceKey: connectionName1, settings1);
+        builder.AddKeyedRavenDBClient(serviceKey: connectionName2, settings2);
+
+        using var host = builder.Build();
+
+        using var documentStore1 = host.Services.GetRequiredKeyedService<IDocumentStore>(connectionName1);
+        using var documentStore2 = host.Services.GetRequiredKeyedService<IDocumentStore>(connectionName2);
+
+        Assert.NotNull(documentStore1);
+        Assert.NotNull(documentStore2);
+
+        using var session1 = host.Services.GetRequiredKeyedService<IDocumentSession>(connectionName1);
+        using var session2 = host.Services.GetRequiredKeyedService<IDocumentSession>(connectionName2);
+
+        Assert.Equal(DefaultConnectionString, documentStore1.Urls[0]);
+        Assert.Equal(DefaultConnectionString, documentStore2.Urls[0]);
+
+        Assert.Equal(databaseName1, documentStore1.Database);
+        Assert.Equal(databaseName2, documentStore2.Database);
+
+        Assert.NotEqual(session1, session2);
+
+        var product = session1.Load<object>("products/77-A");
+        Assert.Null(product);
+
+        product = session2.Load<object>("products/77-A");
+        Assert.Null(product);
+    }
+
+    [Fact]
+    public void AddRavenDbClientShouldWork_Create2NewDatabases()
+    {
+        var databaseName1 = Guid.NewGuid().ToString("N");
+        var databaseName2 = Guid.NewGuid().ToString("N");
+
+        var connectionName1 = Guid.NewGuid().ToString("N");
+        var connectionName2 = Guid.NewGuid().ToString("N");
+
+        IEnumerable<KeyValuePair<string, string?>> config =
+        [
+            new KeyValuePair<string, string?>($"ConnectionStrings:{connectionName1}", DefaultConnectionString),
+            new KeyValuePair<string, string?>($"ConnectionStrings:{connectionName2}", DefaultConnectionString)
+        ];
+        var builder = CreateBuilder(config);
+
+        var settings1 = GetDefaultSettings(databaseName1);
+        var settings2 = GetDefaultSettings(databaseName2);
+
+        builder.AddRavenDBClient(settings1);
+        builder.AddRavenDBClient(settings2);
+
+        using var host = builder.Build();
+
+        using var documentStore1 = host.Services.GetRequiredService<IDocumentStore>();
+        using var documentStore2 = host.Services.GetRequiredService<IDocumentStore>();
+
+        Assert.NotNull(documentStore1);
+        Assert.NotNull(documentStore2);
+
+        using var session1 = host.Services.GetRequiredService<IDocumentSession>();
+        using var session2 = host.Services.GetRequiredService<IDocumentSession>();
+
+        Assert.Equal(DefaultConnectionString, documentStore1.Urls[0]);
+        Assert.Equal(DefaultConnectionString, documentStore2.Urls[0]);
+
+        Assert.True(documentStore1.Database == databaseName1 || documentStore1.Database == databaseName2);
+        Assert.True(documentStore2.Database == databaseName1 || documentStore2.Database == databaseName2);
```

Comment:
Why would the store be one of two databases? Shouldn't that be a known assert?

Reply:
Removed